### ISSUE TITLE
Volume attachment missing tagging support for virtual device identification

### ISSUE THREAD

**Issue by vveliev-tc** [terraform-provider-openstack/terraform-provider-openstack#1262]
I'm trying to use this OpenStack provider to attach volumes to existing VMs (compute instance v2) 

However I'm not able to get OpenStack metadata to include information about attached devices, this is required for data persistence 

according to API this is supported https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/10/html/networking_guide/use-tagging

### Terraform Version
terraform 14.6 

### Affected Resource(s)
- openstack_compute_volume_attach_v2

### Terraform Configuration Files
```hcl
resource "openstack_compute_volume_attach_v2" "attached_list" {
  count       = length(var.attach_compute_instance_list)
  instance_id = var.attach_compute_instance_list[count.index]
  volume_id   = openstack_blockstorage_volume_v3.volume_list[count.index].id
}

```

### Expected Behavior
Expecting to have a deterministic way to find the device name that is attached to the server

```sh
openstack server add volume --tag asdf test_tag test
```

```
curl $host/openstack/latest/meta_data.json
```

should contain
```
 "devices": [{"type": "disk", "serial": "8655fe03-dd5a-4130-85c2-341f9eeac808", "bus": "scsi", "address": "none", "tags": ["asdf"]}]
```

### Actual Behavior

I was not able to figure out how to attach the volume to an already running server with the ability to deterministically guarantee that devices name and logical mount (particularly useful when have multiple attach volumes)

**Comment by nikParasyr** [comment]
@vveliev-tc Thanks for reporting this. 

For future reference, when providing doc urls please use docs of the vanilla/plain openstack and not redhat( or any other vendor for this matter).

There is quite some things mentioned here so i just want to clarify that i understand it right: 
Basically you are missing the ability to add a tag on an attachment. This is possible based on [docs](https://docs.openstack.org/api-ref/compute/?expanded=attach-a-volume-to-an-instance-detail#attach-a-volume-to-an-instance). If this is the case we need to update gophercloud first as it doesn't support it yet. I'll open a ticket there

**Comment by vveliev-tc** [comment]
> There is quite some things mentioned here so i just want to clarify that i understand it right:
> Basically you are missing the ability to add a tag on an attachment. This is possible based on [docs](https://docs.openstack.org/api-ref/compute/?expanded=attach-a-volume-to-an-instance-detail#attach-a-volume-to-an-instance). If this is the case we need to update gophercloud first as it doesn't support it yet. I'll open a ticket there

Yes I think that would be the native way to attache volumes persistently without rebooting, as you guys mentioned in https://github.com/gophercloud/gophercloud/issues/2176 it's possible to tag in when creating VM but not to existing VM

**Comment by nikParasyr** [comment]
@vveliev-tc this has been implemented upstream. Currently I'm focusing on fixing the test environment and some other changes that have higher priority (for me), after these ill have a look implementing this as well. If you want it implemented asap feel free to make a PR regarding this, it should be quite straight forward.
